Modules/_sqlite/connection.c

```diff
@@ -432,48 +437,81 @@ free_callback_contexts(pysqlite_Connection *self)
 static void
 remove_callbacks(sqlite3 *db)
 {
-    sqlite3_trace_v2(db, SQLITE_TRACE_STMT, 0, 0);
+    /* None of these APIs can fail, as long as they are given a valid
+     * database pointer. */
+    assert(db != NULL);
+    int rc = sqlite3_trace_v2(db, SQLITE_TRACE_STMT, 0, 0);
+    assert(rc == SQLITE_OK), (void)rc;
+
     sqlite3_progress_handler(db, 0, 0, (void *)0);
-    (void)sqlite3_set_authorizer(db, NULL, NULL);
+
+    rc = sqlite3_set_authorizer(db, NULL, NULL);
+    assert(rc == SQLITE_OK);
 }
 
-static void
+static int
 connection_close(pysqlite_Connection *self)
 {
+    int rc = 0;
     if (self->db) {
```

Comment:
Maybe use ``if (self->db == NULL) { return 0; }`` to avoid one indentation level.

Reply:
Sure, let's do that now that we're touching this code anyways.